AlassaneWone/RoomMapping
# TT27- End2End testing

Implemented two end-to-end tests using Cypress to ensure the functionality of the application.

=== Test 1: Connection Flow
- Simulates the user's interaction with the login flow.
- Visits the home page, clicks on "Se connecter," fills in the login form, and verifies successful redirection to the home page after login.

=== Test 2: User Map Display
- Verifies the correct display of maps for the logged-in user on the MapList page.
- Intercepts the API call for user maps, stubs it with test data, and checks if the maps are displayed correctly.

=== Running Tests with Cypress:
1. Ensure Cypress is installed as a devDependency in package.json:  
   ```bash -> \RoomMapping\Frontend-user-interface  
   npm install

4. Run Cypress to open the Cypress Test Runner: 
    `npx cypress open`

6. In the Cypress Test Runner, click on the respective test file to run the tests.

This commit message provides a brief description of the implemented tests and includes instructions on installing and running Cypress for end-to-end testing.  

For more information about these tests : https://github.com/AlassaneWone/RoomMapping/wiki/9.-Testing#tests-end2end  
[Cypress](https://www.cypress.io/)

TT27-end2end_testing
